src/include/miopen/conv/problem_description.hpp

```diff
@@ -151,16 +153,16 @@ struct ProblemDescription : ProblemDescriptionBase
 {
     ProblemDescription() = default;
 
-    ProblemDescription(const TensorDescriptor& in_,
-                       const TensorDescriptor& weights_,
-                       const TensorDescriptor& out_,
-                       const ConvolutionDescriptor& conv_,
+    ProblemDescription(TensorDescriptor in_,
+                       TensorDescriptor weights_,
+                       TensorDescriptor out_,
+                       ConvolutionDescriptor conv_,
```

Comment:
The arguments are copied when passed by value, in this case using `std::move` has no advantages

Reply:
With this pattern they can be either copied or moved from outside.

Reply:
`TensorDescriptor` and `ConvolutionDescriptor` both contains a few small vectors that could be moved and several variables that are copied anyway. I think that this will not affect the speed, since memory allocation here takes much longer than copying.

Reply:
I think in this case it's worth adding a separate constructor that takes rvalue references